src/StoragelessSession/Session/SessionScope.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+declare(strict_types = 1);
+
+namespace StoragelessSession\Session;
+
+class SessionScope
+{
+    /**
+     * @var \DateTimeImmutable
+     */
+    private $expirationTime;
+
+    /**
+     * @var bool
+     */
+    private $isModified = false;
+
+    /**
+     * @var mixed[]
+     */
+    private $data = [];
+
+    /**
+     * @param mixed[]                 $value
+     * @param \DateTimeImmutable|null $dateTimeImmutable
+     */
+    public function __construct(array $value, \DateTimeImmutable $dateTimeImmutable = null)
+    {
+        $this->data           = $value;
+        $this->expirationTime = $dateTimeImmutable;
+    }
+
+    public function set(string $key, $value)
+    {
+        $this->data[$key] = $value;
+        $this->isModified = true;
+    }
+
+    public function get(string $key, $default = null)
+    {
+        if (! isset($this->data[$key])) {
+            $this->data[$key] = $default;
+        }
+
+        return $this->data[$key];
+    }
+
+    public function remove(string $key)
+    {
+        unset($this->data[$key]);
+        $this->isModified = true;
+    }
+
+    public function isModified() : bool
+    {
+        return $this->isModified;
+    }
+
+    public function isEmpty() : bool
+    {
+        return empty($this->data);
+    }
+
+    public function getData() : array
+    {
+        return $this->data;
+    }
+
+    public function getExpirationTime()
```

Comment:
Probably not needed. Instead, we need to be able to set the expiration time, and reset the scope (including resetting the expiration time) on any method call